### PR TITLE
IfcConvert enhancements

### PR DESCRIPTION
**Pull request by Stinkfist0**
--precision
--model-offset
--unicode

re: #20 #73 #122

### CODE REVIEW

**Comment by Stinkfist0** [comment]
I could also throw in the --model-offset feature to the same pull request soon or provide a new PR with it, whatever suits you the best.

**Comment by Stinkfist0** [comment]
I could also squeeze in `--unicode=escape|utf8` as it's pretty trivial, I should be done with in within an hour.

**Comment by Stinkfist0** [comment]
OK, that should be it for now.

File: src/ifcconvert/IfcConvert.cpp [hunk 1]
```diff
@@ -482,16 +517,25 @@ int main(int argc, char** argv) {
...
-	}
+    if (center_model || model_offset) {
+        double* offset = serializer->settings().offset;
+        if (center_model) {
+            gp_XYZ center = (context_iterator.bounds_min() + context_iterator.bounds_max()) * 0.5;
+            offset[0] = -center.X();
+            offset[1] = -center.Y();
+            offset[2] = -center.Z();
+        } else {
+            if (sscanf(offset_str.c_str(), "%lf;%lf;%lf", &offset[0], &offset[1], &offset[2]) != 3) {
```
Comment (aothms) [inline src/ifcconvert/IfcConvert.cpp]: Wonder if we need to do something with `setlocale()` here? I'm somewhat surprised that I need to use a dot as a decimal separator even though my OS is set to believe in the mainland European convention of using commas. It's a minor issue and I'm sure users can figure this out for themselves.

**Comment by aothms** [comment]
Wonderful, just waiting for your opinion on the locale thing and then I will merge it.

**Comment by Stinkfist0** [comment]
Hmm, I'm suspecting we never set the locale explicitly anywhere; the default implicit locale for for any C and C++ programs should be (IIRC)  the C locale.

**Comment by aothms** [comment]
Yeah just I had to get too so much trouble to make 100% sure it is the C locale in the parser [1] that I got a bit allergic to this. But I suppose this is mostly for *nix users messing with `LC_ALL` that supposedly know what they are doing. Great!

[1] https://github.com/IfcOpenShell/IfcOpenShell/blob/master/src/ifcparse/IfcParse.cpp#L48

**Pull request merged**